Source/WTF/wtf/glib/FileSystemGlib.cpp

```diff
@@ -134,6 +134,18 @@ std::optional<uint64_t> fileSize(PlatformFileHandle handle)
     return g_file_info_get_size(info.get());
 }
 
+std::optional<PlatformFileID> fileID(PlatformFileHandle handle)
+{
+    // FIXME (246118): Implement this function properly.
+    return std::nullopt;
+}
+
+bool fileIDsAreEqual(std::optional<PlatformFileID> a, std::optional<PlatformFileID> b)
```

Comment:
Is this method needed? Wouldn't `std::nullopt == std::nullopt`, for platforms that don't implement `fileID`?

Reply:
For POSIX and GLIB, `PlatformFileID` is a scalar, and so `operator ==` for `std::optional< PlatformFileID>` compiles fan because it transitively compares the two scalar values. But for Windows, this is not the case, since `PlatformFileID` is a struct, and so it fails to compile.

Reply:
Why don't we use operator== instead? It seems wasteful that this is a full function call in platforms where it's a 64-bit scalar value. We might want to just in-line it in platforms other than Windows.

Reply:
> We might want to just in-line it in platforms other than Windows.

How would I do this?

Reply:
FTR, I have the GLib implementation of `FileSystem::fileID()` in https://github.com/WebKit/WebKit/pull/5081 and I'm using `CString` there as the type of the identifier.

Reply:
@aperezdc ah ok; does Cstring support `operator ==`?

Reply:
> @aperezdc ah ok; does Cstring support `operator ==`?

Yes.